plugins/smartvol/bricksplanner/utils.go

```diff
@@ -0,0 +1,108 @@
+package bricksplanner
+
+import (
+	"encoding/json"
+	"sort"
+	"strings"
+
+	"github.com/gluster/glusterd2/glusterd2/peer"
+	"github.com/gluster/glusterd2/glusterd2/store"
+	"github.com/gluster/glusterd2/pkg/utils"
+	deviceapi "github.com/gluster/glusterd2/plugins/device/api"
+	smartvolapi "github.com/gluster/glusterd2/plugins/smartvol/api"
+)
+
+var subvolPlanners = make(map[string]SubvolPlanner)
+
+// SubvolPlanner represents planner interface for different subvol types
+type SubvolPlanner interface {
+	Init(*smartvolapi.Volume, uint64)
+	BricksCount() int
+	BrickSize(int) uint64
+	BrickType(int) string
+}
+
+// Vg represents Virtual Volume Group
+type Vg struct {
+	Name          string
+	DeviceName    string
+	PeerID        string
+	Zone          string
+	State         string
+	AvailableSize uint64
+	Used          bool
+}
+
+func getAvailableVgs(req *smartvolapi.Volume) ([]Vg, error) {
+	var vgs []Vg
+	peers, err := peer.GetPeers()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, p := range peers {
+		// If Peer is not online, do not consider this device/peer
+		if !store.Store.IsNodeAlive(p.ID) {
+			continue
+		}
+
+		peerzone, exists := p.Metadata["_zone"]
+		if !exists || strings.TrimSpace(peerzone) == "" {
+			peerzone = p.ID.String()
+		}
+
+		// If List of Peer IDs specified to limit choosing the bricks from
+		if len(req.LimitPeers) > 0 && !utils.StringInSlice(p.ID.String(), req.LimitPeers) {
+			continue
+		}
+
+		// If List of zones specified to limit choosing the bricks from
+		if len(req.LimitZones) > 0 && !utils.StringInSlice(peerzone, req.LimitZones) {
+			continue
+		}
+
+		// If Exclude List of Peer IDs specified
+		if len(req.ExcludePeers) > 0 && utils.StringInSlice(p.ID.String(), req.ExcludePeers) {
+			continue
+		}
+
+		// If Exclude List of Zones specified
+		if len(req.ExcludeZones) > 0 && utils.StringInSlice(peerzone, req.ExcludeZones) {
+			continue
+		}
+
+		devicesRaw, exists := p.Metadata["_devices"]
```

Comment:
Once https://github.com/gluster/glusterd2/pull/828 this is merged, you can use function directly to get the devices